packages/web-components/fast-components/src/menu-item/menu-item.styles.ts

```diff
@@ -108,25 +113,151 @@ export const MenuItemStyles = css`
     :host(:active)::slotted(svg) {
         fill: ${accentForegroundCutRestBehavior.var};
     }
+
+    :host .start-container {
+        display: grid;
+        grid-template-columns: auto auto;
+        align-items: center;
+        justify-content: center;
+    }
+
+    :host([role="menuitemcheckbox"]) .start-container,
+    :host([role="menuitemradio"]) .start-container {
+        column-gap: 10px;
+    }
+
+    :host([role="menuitemcheckbox"]),
+    :host([role="menuitemradio"]) {
+        display: grid;
+        grid-template-columns: auto 1fr minmax(42px, auto);
+        align-items: center;
+        min-height: 32px;
+    }
+
+    :host([role="menuitemcheckbox"]) .start,
+    :host([role="menuitemradio"]) .start {
+        grid-column-start: 2;
+        margin-inline-end: 10px;
+    }
+
+    :host([role="menuitemcheckbox"]) .end,
+    :host([role="menuitemradio"]) .end {
+        grid-column-start: 3;
+    }
+
+    :host .checkbox {
+        position: relative;
+        width: 20px;
+        height: 20px;
+        box-sizing: border-box;
+        border-radius: calc(var(--corner-radius) * 1px);
+        border: calc(var(--outline-width) * 1px) solid ${
+            accentForegroundCutRestBehavior.var
+        };
+        outline: none;
+        margin-inline-start: 10px;
+    }
+
+    :host(:hover) .checkbox {
+        border-color: ${accentForegroundCutRestBehavior.var};
+    }
+
+    :host([checked="true"]) .checkbox {
+        display: none;
+    }
+
+    :host([checked="true"]) .checkbox-checked {
+        position: relative;
+        width: 20px;
+        height: 20px;
+        box-sizing: border-box;
+        border-radius: calc(var(--corner-radius) * 1px);
+        border: calc(var(--outline-width) * 1px) solid ${
+            accentForegroundCutRestBehavior.var
+        };
+        background: ${accentForegroundCutRestBehavior.var};
+        outline: none;
+        margin-inline-start: 10px;
+    }
+
+    :host([checked="true"]) .checkbox-indicator{
+        width: 100%;
+        height: 100%;
+        display: block;
+        fill: ${accentFillRestBehavior.var};
+        pointer-events: none;
+    }
+
+    :host .radio {
+        position: relative;
+        width: 20px;
+        height: 20px;
+        box-sizing: border-box;
+        border-radius: 999px;
+        border: calc(var(--outline-width) * 1px) solid ${
+            accentForegroundCutRestBehavior.var
+        };
+        outline: none;
+        margin-inline-start: 10px;
+    }
+
+    :host(:hover) .radio {
+        border-color: ${accentForegroundCutRestBehavior.var};
+    }
+
+    :host([checked="true"]) .radio {
```

Comment:
Can we bind this to the `aria-checked` attribute rather than `checked`? IMO, we likely shouldn't be reflecting `checked` - it should likely only be `fromView` (no need to change in this PR)

Reply:
updated to use aria-checked